code/parse/parselo.cpp

```diff
@@ -2104,10 +2104,9 @@ int parse_get_line(char *lineout, int max_line_len, const char *start, int max_s
 // Goober5000 - added ability to read somewhere other than Parse_text
 void read_file_text(const char *filename, int mode, char *processed_text, char *raw_text)
 {
-	// copy the filename
-	if (!filename)
-		throw parse::ParseException("Invalid filename");
```

Comment:
Is there a reason we shouldn't still throw a parse exception if no filename was given?

Reply:
There shouldn't ever be a case where these functions are passed a NULL filename.  Empty filename perhaps, but a NULL filename is a coding error.

Reply:
I agree, but that doesn't change the situation, I'm not advocating the removal of the assertion. If the filename is null, release builds *definitely* should not attempt to do anything with it and instead throw a parse exception to whatever the outer context is.

Reply:
I'm skeptical that the exception would ever be thrown, but ok.

Reply:
Me too, but I've grown wary of assertions of a thing which should always be true... but then continue anyway if it fails, guaranteeing a crash when a safer early out could have been done instead.